src/cmd/ksh93/bltins/enum.c

```diff
@@ -302,11 +302,11 @@ int b_enum(int argc, char **argv, Shbltin_t *context) {
     }
 
     argv += opt_info.index;
-    if (error_info.errors) {
+    if (error_info.errors || !*argv || *(argv + 1)) {
```

Comment:
This is an atypical, and confusing, way to do the test for the wrong number of arguments. This should be

```
argc -= opt_info.index;
if (error_info.errors || argc != 1) {
```